tests/pr_test.go

```diff
@@ -1,56 +1,264 @@
+	} else {
+
+		options := testschematic.TestSchematicOptionsDefault(&testschematic.TestSchematicOptions{
+			Testing: t,
+			Prefix:  "monitoring-agent",
+			TarIncludePatterns: []string{
+				"*.tf",
+				"kubeconfig/*.*",
+				"scripts/*.*",
+				fullyConfigurableSolutionDir + "/*.*",
+				fullyConfigurableSolutionKubeconfigDir + "/*.*",
+			},
+			ResourceGroup:          resourceGroup,
+			TemplateFolder:         fullyConfigurableSolutionDir,
+			Tags:                   []string{"test-schematic"},
+			DeleteWorkspaceOnFail:  false,
+			WaitJobCompleteMinutes: 60,
+			Region:                 region,
+		})
+
+		options.TerraformVars = []testschematic.TestSchematicTerraformVar{
+			{Name: "ibmcloud_api_key", Value: options.RequiredEnvironmentVars["TF_VAR_ibmcloud_api_key"], DataType: "string", Secure: true},
+			{Name: "cloud_monitoring_instance_region", Value: region, DataType: "string"},
+			{Name: "cluster_id", Value: terraform.Output(t, existingTerraformOptions, "cluster_id"), DataType: "string"},
+			{Name: "cluster_resource_group_id", Value: terraform.Output(t, existingTerraformOptions, "cluster_resource_group_id"), DataType: "string"},
+			{Name: "cloud_monitoring_access_key", Value: terraform.Output(t, existingTerraformOptions, "cloud_monitoring_access_key"), DataType: "string", Secure: true},
+			{Name: "prefix", Value: options.Prefix, DataType: "string"},
+		}
+
+		err := options.RunSchematicUpgradeTest()
+		assert.Nil(t, err, "This should not have errored")
+	}
+
+	// Check if "DO_NOT_DESTROY_ON_FAILURE" is set
+	envVal, _ := os.LookupEnv("DO_NOT_DESTROY_ON_FAILURE")
+	// Destroy the temporary existing resources if required
+	if t.Failed() && strings.ToLower(envVal) == "true" {
+		fmt.Println("Terratest failed. Debug the test and delete resources manually.")
+	} else {
+		logger.Log(t, "START: Destroy (existing resources)")
+		terraform.Destroy(t, existingTerraformOptions)
+		terraform.WorkspaceDelete(t, existingTerraformOptions, prefix)
+		logger.Log(t, "END: Destroy (existing resources)")
+	}
+}
+
+func TestRunAgentVpcKubernetes(t *testing.T) {
+	t.Parallel()
+
+	options := setupOptions(t, "obs-agent-iks", terraformDirMonitoringAgentIKS)
 	output, err := options.RunTestConsistency()
 	assert.Nil(t, err, "This should not have errored")
 	assert.NotNil(t, output, "Expected some output")
 }
 
-func TestRunAdvancedExample(t *testing.T) {
+func TestRunAgentVpcOcp(t *testing.T) {
 	t.Parallel()
 
-	options := setupOptions(t, "mod-template-adv", advancedExampleDir)
-
+	options := setupOptions(t, "obs-agent-roks", terraformDirMonitoringAgentROKS)
 	output, err := options.RunTestConsistency()
 	assert.Nil(t, err, "This should not have errored")
 	assert.NotNil(t, output, "Expected some output")
 }
 
-// Upgrade test (using advanced example)
-func TestRunUpgradeExample(t *testing.T) {
+func TestRunAgentVpcOcpUpgrade(t *testing.T) {
```

Comment:
this is not needed - we already have an upgrade test with the DA. I would replace this with `TestRunAgentClassicKubernetes` from other_test.go